imaging/ml/ml_codelab/scripts/inference/inference.py

```diff
@@ -65,7 +65,7 @@
 from email.mime import multipart
 import json
 import logging
-import mimetools
+import email.generator
```

Comment:
@codezizo As you know, this library has caused us problems with Python3 :)
I'd expect this to be an issue here as well?

Reply:
@mfomitchev yes that is true. @pavertomato  can you please avoid using: 

from email.mime import multipart
multipart.MIMEMultipart

and instead use
```
import urllib3

    parts = []
    application_type = 'dicom+json'
    jsonstr = json.dumps(input_data)
    part = urllib3.fields.RequestField(
        name='placeholder',
        data=jsonstr,
        headers={'Content-Type': 'application/%s' % application_type},
    )
    parts.append(part)
    boundary=email.generator._make_boundary()
    body, _ = urllib3.filepost.encode_multipart_formdata(parts, boundary)
```

Reply:
When we are done with this, would be good to do a quick end-to-end test at the end to make sure everything still works.

Reply:
>caused us problems with Python3 

not actually understand what problems, email.generator works with py2 and py3, or it's about some special env? but this code is running inside of docker image and env is constant
![image](https://user-images.githubusercontent.com/878990/71368372-5eb15680-25da-11ea-83f5-f6d0cde52018.png)
>and instead use

i don't actually understand, in your example you still use email.generator
>When we are done with this, would be good to do a quick end-to-end test at the end to make sure everything still works.

i've done manual testing, current version works, but for e2e testing would be great to separate ml examples to new repo, or we would have to create cloud build triggers for healthcare repo just for this small tool

Reply:
The problem is not emal.generator the problem is from email.mime import multipart

Reply:
i've updated way of request generation and manually tested it
but i don't know what kind of problem is mentioned, previous version worked well too

Reply:
One of the problems I saw was that the Py3 email mime encoder would actually replace all standalone '\n' chars with '\r\n'.. which is obv a problem with binary content. I think @codezizo ran into some other issues with it as well, although I don't remember the specifics anymore. It would run fine, but then if you tried viewing the stored DICOM instance in eUnity, it wouldn't show up correctly.

Reply:
oh, thanks,
well, in this case my recent commit may be reverted without consequences because this code sends only text data(json)

Reply:
Ah yes, good point. I'll leave it up to @codezizo whether to land the commit. Eventually we plan to create a separate 'inference toolkit' library (which would include StowRs among other things) and have the codelab just use that, so IMHO doesn't matter all that much what we do here as long as it works. Either way, we should do a full end-to-end test once we are done with all the code changes.

Reply:
>Eventually we plan to create a separate 'inference toolkit' library(which would include StowRs among other things)

interesting coincidence what recently has been developed cli python tool to simplify work with dicomweb, and since target is quite close, i think this cli tool may be split into cli + library and library may become more common and reusable, repo with this tool is not public yet, but i think @dzelemba may give you access?

Reply:
Yeah, I'd be interested in taking a look.
My guess is that the overlap might not be that large in this case, but we also have something else in the pipe which might be a better candidate for reusing the code there.

Reply:
Since we are only creating text json, and if it is working. Then it is okay as it is. As we do plan on migrating to use a different library once it is published. Thanks for the thorough explanations.

Reply:
so should i revert this commit https://github.com/GoogleCloudPlatform/healthcare/pull/426/commits/018cbe5123bccd68b87d3502567660fbce8c7759 or keep current state of branch?

Reply:
Yes please feel free to revert. Sorry about the confusion.

Reply:
done